src/server/game/VoiceChat/VoiceChatMgr.cpp

```diff
@@ -0,0 +1,1385 @@
+            }
+            else
+                LOG_ERROR("voice-chat", "Failed to create voice chat session or socket is not open");
+
+            // Connection lost, clean up and prepare for reconnect
+            if (_enabled) // Only set reconnecting state if still enabled
+            {
+                LOG_INFO("voice-chat", "Preparing for reconnection...");
+                HandleConnectionLoss();
+            }
+        }
+        catch (boost::system::system_error const& e)
+        {
+            LOG_ERROR("voice-chat", "Boost system error in voice chat thread: {}", e.what());
+            if (_enabled)
+                HandleConnectionFailure();
+        }
+        catch (std::exception const& e)
+        {
+            LOG_ERROR("voice-chat", "Exception in voice chat thread: {}", e.what());
+            if (_enabled)
+                HandleConnectionFailure();
+        }
+        catch (...)
+        {
+            LOG_ERROR("voice-chat", "Unknown exception in voice chat thread");
+            if (_enabled)
+                HandleConnectionFailure();
+        }
+
+        // Small delay before next iteration
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    }
+
+    LOG_INFO("voice-chat", "Voice socket thread exiting");
+}
+
+void VoiceChatMgr::LoadConfigs()
+{
+    _enabled = sWorld->getBoolConfig(CONFIG_VOICE_CHAT_ENABLED);
+
+    _serverAddressString = sConfigMgr->GetOption<std::string>("VoiceChat.ServerAddress", "127.0.0.1");
+    _serverAddress = inet_addr(_serverAddressString.c_str());
+    _serverPort = sWorld->getIntConfig(CONFIG_VOICE_CHAT_SERVER_PORT);
+
+    std::string voice_address_string = sConfigMgr->GetOption<std::string>("VoiceChat.VoiceAddress", "127.0.0.1");
+    _voiceAddress = inet_addr(voice_address_string.c_str());
+    _voicePort = sWorld->getIntConfig(CONFIG_VOICE_CHAT_VOICE_PORT);
+
+    _maxConnectAttempts = sWorld->getIntConfig(CONFIG_VOICE_CHAT_MAX_CONNECT_ATTEMPTS);
+}
+
+void VoiceChatMgr::Init()
+{
+    LoadConfigs();
+
+    _nextPing = std::chrono::system_clock::now() + std::chrono::seconds(5);
+    _lastPong = std::chrono::system_clock::now();
+    _lastUpdate = std::chrono::system_clock::now();
+    _curReconnectAttempts = 0;
+
+    if (_enabled)
+    {
+        _state = VOICECHAT_NOT_CONNECTED;
+        _nextConnect = std::chrono::system_clock::now();
+
+        // Auto-start the voice chat connection thread
+        LOG_INFO("voice-chat", "Voice chat is enabled, starting connection thread");
+        ActivateVoiceSocketThread();
+    }
+    else
+    {
+        _state = VOICECHAT_DISCONNECTED;
+        LOG_INFO("voice-chat", "Voice chat is disabled");
+    }
+}
+
+void VoiceChatMgr::Update()
+{
+    if (!_enabled)
+        return;
+
+    LOG_DEBUG("voice-chat", "VoiceChatMgr::Update state: {}", _state);
+
+    _eventEmitter(this);
+
+    auto currentSocket = _socket.lock();
+
+    if (currentSocket)
+        currentSocket->Update();
```

Comment:
Why are we calling socket update from the `World` thread when you've introduced a separate thread for this socket?